### PR TITLE
extended: set failure traps in all deployment tests

### PR DESCRIPTION
**Pull request by kargakis**
@mfojtik ptal

@smarterclayton fyi

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
secrets flake and clean conformance - @mfojtik we need this in order to start debugging any remanining deployment flakes

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
Second clean conformance run - @mfojtik let's get this asap merged

**Comment by mfojtik** [comment]
lgtm [merge]

File: test/extended/deployments/deployments.go [hunk 1]
```diff
@@ -451,6 +570,23 @@ var _ = g.Describe("deploymentconfigs", func() {
 	})
 
 	g.Describe("rolled back", func() {
+		g.AfterEach(func() {
```
Comment (smarterclayton) [inline test/extended/deployments/deployments.go]: This is pretty close to the point where it needs to be abstracted - you added a few hundred lines of duplication.

Comment (kargakis) [inline test/extended/deployments/deployments.go]: I hope it's better now

**Comment by kargakis** [comment]
conformance hung [merge]

**Comment by kargakis** [comment]
```
FAILURE after 16.854s: test/end-to-end/core.sh:76: executing 'docker pull centos/ruby-22-centos7:latest' expecting success: the command returned the wrong error code
<>
failed to register layer: ApplyLayer open /dev/mapper/docker-202:2-67676390-5bb0ce60ec5ae7840645477c212e0b7cb3084adf1a6dea123b1e37a01984e6e7: no such file or directory stdout: {"layerSize":0}
```
[merge]

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**